src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -142,6 +142,8 @@ trait Blockchain {
 
   def getBestBlockNumber(): BigInt
 
+  def getBestSavedBlockNumber(): BigInt
```

Comment:
I'm wondering what are the repercussions of `getBestBlockNumber()` failing.
I have experienced an issue when calling `getBestBlock()` and getting `None` as a result. `getBestBlock()` calls `getBestBlockNumber()`.
Should `getBestBlockNumber()` be adjusted, instead of creating a new `getBestSavedBlockNumber()`?

Reply:
we need `getBestBlockNumber` as is, as some methods require to know best known number (even though it may be not stored yet. I adjusted `getBestBlock()` instead to fall back to the best stored block in case best known is not available. 

Will push in a bit after I'm done fixing tests